bigtable-hbase-1.x-parent/bigtable-hbase-1.x/src/test/java/com/google/cloud/bigtable/hbase1_x/TestMetrics.java

```diff
@@ -204,39 +224,64 @@ public void appendFailure() throws IOException {
       return (T) requests.poll();
     }
 
+    public synchronized long getReadRowServerSideLatency() {
+      return readRowServerSideLatency;
+    }
+
+    public synchronized long getMutateRowServerSideLatency() {
+      return mutateRowServerSideLatency;
+    }
+
+    public synchronized long getReadModifyWriteRowServerSideLatency() {
+      return readModifyWriteRowServerSideLatency;
+    }
+
     @Override
     public void readRows(
         ReadRowsRequest request, StreamObserver<ReadRowsResponse> responseObserver) {
       requests.add(request);
 
-      if (callCount.getAndIncrement() < 4) {
-        responseObserver.onError(new StatusRuntimeException(Status.UNAVAILABLE));
-      } else {
-        try {
-          Thread.sleep(20);
-        } catch (InterruptedException e) {
-          e.printStackTrace();
+      synchronized (lock) {
+        if (!readRowsStopwatch.isRunning()) {
+          readRowsStopwatch.start();
+        }
+        if (callCount.getAndIncrement() < 4) {
+          responseObserver.onError(new StatusRuntimeException(Status.UNAVAILABLE));
+        } else {
+          try {
+            Thread.sleep(20);
+          } catch (InterruptedException e) {
+            e.printStackTrace();
```

Comment:
I would try to swallow these errors. Instead you should bubble them back to the caller:

responseObserver.onError(e);
return;